cmd/amazon-cloudwatch-agent/amazon-cloudwatch-agent.go

```diff
@@ -241,6 +241,8 @@ func runAgent(ctx context.Context,
 	c := config.NewConfig()
 	c.OutputFilters = outputFilters
 	c.InputFilters = inputFilters
+	c.AllowUnusedFields = true
```

Comment:
what is this change for?

Reply:
`start-amazon-cloudwatch-agent` is responsible for a few things: running the `config-translator` binary to generate the TOML/YAML, processing the `run_as_user` to potentially switch users, and starting the `amazon-cloudwatch-agent` binary.

To determine the `run_as_user`, `start-amazon-cloudwatch-agent` currently merges the available JSONs and then parses the JSON to read the field. This is work that is already done by the `config-translator` before it writes the TOML/YAML. This change adds `run_as_user` under the `agent` section of the Telegraf configuration so the `start-amazon-cloudwatch-agent` no longer needs to merge the JSONs to read the file, which avoids repeated work and unnecessary dependencies.

Telegraf normally validates that all fields in the TOML configuration are known fields and used by their components.
`run_as_user` is not a standard field. Setting `AllowUnusedFields` will tell the Telegraf configuration unmarshaler to not fail based on the presence of this unknown field.

Reply:
This is effectively turning off the check entirely, so we do run the risk of adding new unused fields in the future and not getting flagged - although Im not sure why that is so bad. Id imagine any unused fields wont cause any issues, just that they are present for no reason.

Alternative might be to just add `run_as_user` to some allow list like https://github.com/aws/telegraf/blob/483bd77d26a2900938ad835d286a719c5efe954c/config/config.go#L1838?